hardware/esp8266com/esp8266/cores/esp8266/core_esp8266_si2c.c

```diff
@@ -42,35 +39,35 @@ static unsigned char twi_sda, twi_scl;
 #define TWI_CLOCK_STRETCH 1600
 #endif
 
-void twi_setClock(unsigned int freq){
+void twi_setClock(TwiConfig* twi, unsigned int freq){
 #if F_CPU == FCPU80
-  if(freq <= 100000) twi_dcount = 19;//about 100KHz
-  else if(freq <= 200000) twi_dcount = 8;//about 200KHz
-  else if(freq <= 300000) twi_dcount = 3;//about 300KHz
-  else if(freq <= 400000) twi_dcount = 1;//about 400KHz
-  else twi_dcount = 1;//about 400KHz
+  if(freq <= 100000) twi->dcount = 19;//about 100KHz
+  else if(freq <= 200000) twi->dcount = 8;//about 200KHz
+  else if(freq <= 300000) twi->dcount = 3;//about 300KHz
+  else if(freq <= 400000) twi->dcount = 1;//about 400KHz
+  else twi->dcount = 1;//about 400KHz
 #else
-  if(freq <= 100000) twi_dcount = 32;//about 100KHz
-  else if(freq <= 200000) twi_dcount = 14;//about 200KHz
-  else if(freq <= 300000) twi_dcount = 8;//about 300KHz
-  else if(freq <= 400000) twi_dcount = 5;//about 400KHz
-  else if(freq <= 500000) twi_dcount = 3;//about 500KHz
-  else if(freq <= 600000) twi_dcount = 2;//about 600KHz
-  else twi_dcount = 1;//about 700KHz
+  if(freq <= 100000) twi->dcount = 32;//about 100KHz
+  else if(freq <= 200000) twi->dcount = 14;//about 200KHz
+  else if(freq <= 300000) twi->dcount = 8;//about 300KHz
+  else if(freq <= 400000) twi->dcount = 5;//about 400KHz
+  else if(freq <= 500000) twi->dcount = 3;//about 500KHz
+  else if(freq <= 600000) twi->dcount = 2;//about 600KHz
```

Comment:
Frequencies will likely change given that static `twi_sda`, `twi_scl` and `twi_dcount` are replaced with a pointer argument which needs to be dereferenced. Could you please check how this change affects the clock frequencies?